src/components/MapView/Analyser/index.tsx

```diff
@@ -95,6 +104,24 @@ function Analyser({ extent, classes }: AnalyserProps) {
   const [aboveThreshold, setAboveThreshold] = useState('');
   const [thresholdError, setThresholdError] = useState<string | null>(null);
 
+  // for polygon intersection analysis
+  const [adminLevel, setAdminLevel] = useState<AdminLevelType>(1);
+  const [startDate, setStartDate] = useState<number | null>(null);
+  const [endDate, setEndDate] = useState<number | null>(null);
+
+  // get variables derived from state
+  const hazardDataType: HazardDataType | null =
+    hazardLayerId === undefined
+      ? null
+      : (LayerDefinitions[hazardLayerId] as WMSLayerProps).geometry ||
+        RasterType.Raster;
+  const hazardServerLayerName = hazardLayerId
+    ? (LayerDefinitions[hazardLayerId] as WMSLayerProps).serverLayerName
+    : undefined;
+  const availableHazardDates = hazardServerLayerName
+    ? availableDates[hazardServerLayerName]?.map(d => new Date(d)) || []
+    : undefined;
```

Comment:
Let's try to reuse / augment `getPossibleDatesForLayer`

Reply:
done

Reply:
good idea.